tiny_dnn/layers/layer.h

```diff
@@ -109,10 +110,34 @@ class layer : public node {
     bool parallelize() const { return parallelize_; }
     bool initialized() const { return initialized_; }
 
+    // TODO(edgar): Deprecated: use the below method 
     core::backend_t backend_type() const {
         return backend_->type();
     }
 
+    core::backend_t backend_type2() const {
```

Comment:
```backend_type2``` is bad name to use... Because backend concept is really new, I think we can just replace old version.